sql/rpl_mi.cc

```diff
@@ -569,9 +570,12 @@ file '%s')", fname);
 	  goto errwithmsg;
 
       /* Starting from 5.5 the master_retry_count may be in the repository. */
-      if (lines >= LINE_FOR_MASTER_RETRY_COUNT &&
-	  init_strvar_from_file(buf, sizeof(buf), &mi->file, ""))
-	  goto errwithmsg;
+      if (lines >= LINE_FOR_MASTER_RETRY_COUNT)
+      {
+        if (init_strvar_from_file(buf, sizeof(buf), &mi->file, ""))
+          goto errwithmsg;
+        mi->retry_count = atol(buf);
```

Comment:
I say it's better to change `init_intvar_from_file`’s `atoi` to `atoll`.

Reply:
recorded at [MDEV-36141](https://jira.mariadb.org/browse/MDEV-36141) for assessment